models/_sector/nft/trades/ethereum/platforms/archipelago_ethereum_base_trades.sql

```diff
@@ -0,0 +1,78 @@
+{{ config(
+    schema = 'archipelago_ethereum',
+    alias ='base_trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_number','tx_hash','sub_tx_trade_id'],
+    )
+}}
+{% set project_start_date='2022-6-20' %}
+
+WITH
+trade_events as (
+    SELECT * FROM {{ source('archipelago_ethereum','ArchipelagoMarket_evt_Trade') }}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
```

Comment:
minor nit:
i could see this going one of two ways:
- remove the non-incremental filter altogether, as this reads from decoded table, so it'll always be at project start date anyway -- although i've found it useful to have to easily apply small data timeframe filters for dev/debugging, i.e. just update the project start date to now() - 1 week or something
- if we keep, change from `endif`, `if` to just `else` for simplicity -- again, minor, but something i've found more readable when building new models

Reply:
I agree "if incremental" -> "else" -> "endif" is the preferred way. :+1: